File: leetcode/1.two-sum.py
#
# @lc app=leetcode id=1 lang=python3
#
# [1] Two Sum
#

# @lc code=start
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        map_count = {}

        for i, num in enumerate(nums):
            compliment = target - num
            # count  = map_count.get(compliment, 0) + 1
            if compliment in map_count:
                return [i, map_count[compliment]]
            else:
                map_count[num] = i


# @lc code=end

File: leetcode/1.two-sum.js
/*
 * @lc app=leetcode id=1 lang=javascript
 *
 * [1] Two Sum
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const basket = {}
    let head = 0
    let tail = 0
    for (let i=0; i<nums.length; i++) {
        const check_other_number = target - nums[i];
        if (basket.hasOwnProperty(check_other_number)) {
            head = basket[check_other_number]
            tail = i
        } else {
            basket[nums[i]]=i
        }
    }

    return [head, tail]
};
// @lc code=end

